services/src/main/scala/cromwell/pipeline/service/ProjectFileService.scala

```diff
@@ -28,7 +28,16 @@ class ProjectFileService(
   ): Future[Either[VersioningException, UpdateFiledResponse]] =
     projectService
       .getUserProjectById(projectId, userId)
-      .flatMap(project => projectVersioning.updateFile(project, projectFile, version))
+      .flatMap(
+        project =>
+          projectVersioning.updateFile(project, projectFile, version).map {
+            case Right((response, version)) => {
+              projectService.updateProjectVersion(version, projectId, userId)
+              Right(response)
```

Comment:
`projectService.updateProjectVersion` returns `Future`. Which means it's possible that this future will fail, but the client will never know about it and will receive successful response. Here's what you should do:
1) Write a test that reproduces this case
2) Make sure it fails
3) Fix `ProjectFileService`
4) Make sure test works